### DISCUSSION TITLE
Generating files

### DISCUSSION THREAD

**Discussion by radekgrzywacz** [nvim-java/nvim-java#307] [Q&A]
Hi! 
I can't find, if this plugin has the functionality to generate classes, interfaces, enums in a intellIj like way? I know that elmcgill/springboot-nvim plugin have it. Do we have our version? If not is it possible to use these both plugins at once?

**Comment by h678288** [comment]
You can install springboot-nvim and add nvim-java as a dependency instead of jdtls. Like this:
![2024-12-06-232348_hyprshot](https://github.com/user-attachments/assets/a6cacab8-da07-4e88-8def-b9091aad2e9d)
Then to make the plugin register non springboot projects you can edit the plugin a little. Go to ~/.local/share/nvim/lazy/springboot-nvim/lua/springboot-nvim, and open the utils.lua file. In the file you will find a function called "find_main_application_class_directory". Edit it to look like this: 
![image](https://github.com/user-attachments/assets/4afa23c2-5551-4e87-a16d-d2748a9865b2)
Now you can setup keybindings for generating new classes, enums and interfaces in java projects. Here are mine for referance, with a function so they are only available when jdtls is attached to the buffer . Its missing some keymaps, made it today so i havent finished.
![image](https://github.com/user-attachments/assets/1a699eff-c6c9-4293-b195-8042cc09dfad)
A bit late but i hope you still find it useful!

**Comment by alessio-vivaldelli** [comment]
I know this thread is old, but I had the same need and I created a separate plugin from this one that serves only that purpose.
If you want, you can find it here:
https://github.com/alessio-vivaldelli/java-creator-nvim
